src/cpu/testers/dr_trace_player/trace_player.cc

```diff
@@ -167,24 +165,59 @@ DRTracePlayer::executeMemInst(DRTraceReader::TraceRef &mem_ref)
 bool
 DRTracePlayer::trySendMemRef(DRTraceReader::TraceRef &mem_ref)
 {
-    PacketPtr pkt = getPacket(mem_ref);
+    PacketPtr pkt, split_pkt;
+
+    std::tie(pkt, split_pkt) = getPacket(mem_ref);
+
+    if (!retrySplitPkt) {
+        // we should not be sending the first pkt again if
+        // we are retrying only on the second part of the
+        // previously stalled request
+        DPRINTF(DRTrace, "Trying to send %s\n", pkt->print());
+
+        if (!port.sendTimingReq(pkt)) {
+            DPRINTF(DRTrace, "Failed to send pkt\n");
+            if (stats.memStallStart == 0) {
+                stats.memStallStart = curTick();
+            }
+            numOutstandingMemReqs++;
```

Comment:
Why do we increment the outstanding requests when the packet is failed?

Reply:
sorry, I pushed the wrong version earlier